linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_dir/ansible/shared.yml

```diff
@@ -6,3 +6,30 @@
 
 {{{ ansible_pam_faillock_enable() }}}
 {{{ ansible_pam_faillock_parameter_value("dir", "var_accounts_passwords_pam_faillock_dir") }}}
+
+- name: '{{{ rule_title }}} - Ensure necessary SELinux packages are installed'
+  ansible.builtin.package:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - python3-libselinux
+    - python3-policycoreutils
+    - policycoreutils-python-utils
+
+- name: '{{{ rule_title }}} - Create the tally directory if it does not exist'
+  ansible.builtin.file:
+    path: "{{ var_accounts_passwords_pam_faillock_dir }}"
+    state: directory
+    setype: 'faillog_t'
+
+- name: '{{{ rule_title }}} - Ensure SELinux file context is permanently set'
+  ansible.builtin.command:
+    cmd: semanage fcontext -a -t faillog_t "{{ var_accounts_passwords_pam_faillock_dir }}(/.*)?"
```

Comment:
Hi, the AutoMatus test fails with Ansible remediations. Bash remediations are OK but Ansible fails. It doesn't happen when a VM backend is used:

```
[jcerny@thinkpad scap-security-guide{pr/9466}]$ python3 tests/automatus.py rule --remediate-using ansible   --libvirt qemu:///system ssgts_rhel9 accounts_passwords_pam_faillock_dir
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-11-08-0826/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_dir
INFO - Script conflicting_settings_authselect.fail.sh using profile (all) OK
INFO - Script expected_faillock_conf.pass.sh using profile (all) OK
INFO - Script expected_pam_files.pass.sh using profile (all) OK
INFO - Script missing_dir_in_authfail.fail.sh using profile (all) OK
INFO - Script missing_dir_in_preauth.fail.sh using profile (all) OK
INFO - Script wrong_faillock_conf.fail.sh using profile (all) OK
INFO - Script wrong_pam_files.fail.sh using profile (all) OK
```

However, on GitHub where we run tests in containers you can find this in the logs (taken from the CS 9 job):

```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "cmd": [
        "semanage",
        "fcontext",
        "-a",
        "-t",
        "faillog_t",
        "/var/log/faillock(/.*)?"
    ],
    "delta": "0:00:00.175286",
    "end": "2022-11-07 15:43:33.863710",
    "invocation": {
        "module_args": {
            "_raw_params": "semanage fcontext -a -t faillog_t \"/var/log/faillock(/.*)?\"",
            "_uses_shell": false,
            "argv": null,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "stdin_add_newline": true,
            "strip_empty_ends": true,
            "warn": false
        }
    },
    "msg": "non-zero return code",
    "rc": 1,
    "start": "2022-11-07 15:43:33.688424",
    "stderr": "ValueError: SELinux policy is not managed or store cannot be accessed.",
    "stderr_lines": [
        "ValueError: SELinux policy is not managed or store cannot be accessed."
    ],
    "stdout": "",
    "stdout_lines": []
}
```

It might be related to the fact htat it's a container. I can see that bash remediation didn't fail, so maybe the bash does something that the Ansible doesn't. I think that we should attempt for consistency between Ansible behavior and Bash behavior. Can you take a closer look?

Reply:
Thanks for the logs. It is indeed because the container is not using SELinux, which makes the task fail and consequently the playbook execution. This failure is also expected in Bash, but the script continues.

Considering the rule is asking about the DIR and not about its SELinux file context, it is fair enough in my point of view to include an `ignore_errors: yes` in this Ansible task. In any case, there is already a specific rule to check the SELinux File Context: `account_password_selinux_faillock_dir`.

I will update the task.

Reply:
amazing